pkg/manager/internal.go

```diff
@@ -459,21 +458,21 @@ func (cm *controllerManager) Start(ctx context.Context) (err error) {
 	// between conversion webhooks and the cache sync (usually initial list) which causes the webhooks
 	// to never start because no cache can be populated.
 	if err := cm.runnables.Webhooks.Start(cm.internalCtx); err != nil {
-		if !errors.Is(err, wait.ErrWaitTimeout) {
+		if !errors.Is(err, context.Canceled) {
 			return err
 		}
 	}
 
 	// Start and wait for caches.
 	if err := cm.runnables.Caches.Start(cm.internalCtx); err != nil {
-		if !errors.Is(err, wait.ErrWaitTimeout) {
+		if !errors.Is(err, context.Canceled) {
```

Comment:
The deprecated message suggest use here wait. Interrupted instead. So, I think we should use it. 
We also could here check if the context was canceled or exceeded.

Reply:
This was following the item 2 in the migration note but I can change to wait.Interrupted

Reply:
but how I understand the code `.Start()` wait until the context is cancelled or there's an error (i.e., no timeouts?) so `context.Canceled` is a better match

Reply:
Should we not either check for `context.DeadlineExceeded`? 

```go
if !errors.Is(err, context.Canceled) && !errors.Is(err, context.DeadlineExceeded)   {

```

Reply:
`cm.internalCtx` has no deadline AFAICS.

Reply:
Consumers of controller-runtime can add their own Runnables to a manager (https://github.com/kubernetes-sigs/controller-runtime/blob/7f57edcf611232b438f2c5700d225c338235f0a4/pkg/manager/runnable_group.go#L53).

We have no control over which wait functions they might use or which errors they return.

I would use the following:
```go
		if !(errors.Is(err, wait.ErrWaitTimeout) || wait.Interrupted(err)) {
			return err
		}
```

`wait.Interrupted` internally checks for the new private version of errWaitTimeout, Canceled and DeadlineExceeded:
```go
func Interrupted(err error) bool {
	switch {
	case errors.Is(err, errWaitTimeout),
		errors.Is(err, context.Canceled),
		errors.Is(err, context.DeadlineExceeded):
		return true
	default:
		return false
	}
}
```

Reply:
@vincepri @alvaroaleman Do you remember why we were continuing with the Start func if we got a wait.ErrWaitTimeout?

Reply:
> `if !(errors.Is(err, wait.ErrWaitTimeout)`

this is also deprecated so it'll trigger a warning

Reply:
That's not a showstopper we can add a no lint (~ `//nolint:<linter-name>`)

Reply:
@mythi Talked to Vince/Alvaro. We would change it to simply return all errors 

Also we would like to merge this PR ASAP and just for now skip the log tests.

Delta looks roughly like this: https://github.com/mythi/controller-runtime/compare/27-upgrade...sbueringer:controller-runtime:pr-27-upgrade-adjusted (I used my branch to verify that everything works as expected with Cluster API)

We will still hold the release until we bumped to 1.27.2 though

Reply:
Sounds good. Done

Reply:
Rebased and included all the proposed updates. We should be getting all tests passing :crossed_fingers:

Reply:
Thx!